lib/AST/TypeJoinMeet.cpp

```diff
@@ -397,7 +397,7 @@ CanType TypeJoin::visitProtocolCompositionType(CanType second) {
 
   // FIXME: Handle other types here.
   if (!First->isExistentialType())
-    return TheAnyType;
+    return Unimplemented;
```

Comment:
Quick question - what does this mean in terms of constraint system? Does it mean that if types can't be joined type variable would get both bindings or should it result in error?

Reply:
It will try both bindings.

Reply:
Thanks!